### DISCUSSION TITLE
deploy on kubernetes with TLS enabled?

### DISCUSSION THREAD

**Discussion by sivaNbalusu5** [halfgaar/FlashMQ#31] [Q&A] [answered]
Hi , I am trying to find documentation for deploying FlashMQ broker on kubenetes as a Pod with tls enabled. Could you please share some notes if you have any?

**Comment by halfgaar** [comment]
I have never used Kubernetes, actually. But, I imagine there are no special steps required.

With Kubernetes, does a load-balancer take care of TLS? In that case, remember the 30000-60000 client limit between two IP addresses. If possible, I'd let FlashMQ do the TLS.

**Reply by sivaNbalusu5** [reply]
May be i did not put the question right....
I want to ask how to pass config file with tls enabled options and where to place the certs while building the docker image?

We dont use load balancer, We use node port to expose 8883(ssl tcp port).

**Reply by halfgaar** [reply, marked as answer]
The readme contains an example docker run command:

```
docker run -p 8883:8883 -v /srv/flashmq/etc/:/etc/flashmq halfgaar/flashmq
```

This example makes the internal `/etc/flashmq` available through the local path `/srv/flashmq/etc`. You can put the config in there, as well as SSL certificates.

I think that's what you mean, once translated to Kubernetes terminology? So you don't put it in the image itself, but in the volume.

On a side note, there is no volume for the db files in this example. If you want restart-persistence, you'll have to make a volume for that too. See https://www.flashmq.org/man/flashmq.conf.5#storage_dir

Same goes for logs.

**Reply by sivaNbalusu5** [reply]
@halfgaar yes, You are right. I have mounted two volumes one for conf and other for storage_dir. everything looks good so far.
